### PR TITLE
PayPal commerce related elements matche Figma design

### LINKED ISSUE #5003: PayPal commerce user facing options should match Figma design
## User Story

As a developer, I want to match Figma design so that user facing options look nice.

## Details
After reviewing Figma design, I find out that few things we can improve:
- The payment options separator is missing.
image

- The credit card icon is missing.
image

- Smart buttons are stacked vertically instead of horizontal.
image

- Button tooltip is missing inadmin setting.
image

## Acceptance Criteria
-   [x] Payment method options style match Figma design

### PR DESCRIPTION
**Pull request by ravinderk**
Resolves #5003

## Description
This PR adds styling fix to match Figma designs. I added a detailed description of styling bugs in the issue description.

## Affects
This pr does not affect exiting pr.

## Visuals
image
VA5uQmCJZw
image

## Pre-review Checklist
-   [x] Acceptance criteria satisfied and marked in a related issue
-   [x] Relevant `@since` tags included in DocBlocks
-   [x] Labels applied if docs are needed
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

## Testing Instructions
Styling fixes added to the frontend for legacy and multi-step form template and admin setting screen, so you can visit both pages to check fixes.

### CODE REVIEW

**Comment by ravinderk** [comment]
As per conversation with @kevinwhoffman  on the issue, I set the smart buttons style layout to  `vertical`: https://github.com/impress-org/givewp/compare/e80a0271c733...84a9166d253a. We can update it later.

File: assets/src/js/frontend/paypal-commerce/AdvancedCardFields.js [hunk 1]
```diff
@@ -21,7 +21,12 @@ class AdvancedCardFields extends PaymentMethod {
 	 */
 	async renderPaymentMethodOption() {
 		if ( ! this.canRenderFields() ) {
-			Array.from( this.form.getElementsByClassName( 'give-paypal-commerce-cc-field-wrap' ) ).forEach( el => el.remove() );
+			Array.from( this.form.getElementsByClassName( 'give-paypal-commerce-cc-field-wrap' ) ).forEach(
+				el => {
+					el.previousElementSibling.remove();
+					el.remove();
+				}
+			);
```
Comment (JasonTheAdams) [inline assets/src/js/frontend/paypal-commerce/AdvancedCardFields.js:29]: What is this doing?

Comment (ravinderk) [reply assets/src/js/frontend/paypal-commerce/AdvancedCardFields.js]: fixed: https://github.com/impress-org/givewp/compare/84a9166d253a...8397a831e245

**Comment by ravinderk** [comment]
> Please fix the input field styling issues. The Cardholder Name field is too tall, big, and has strange shadowing.
> image
> 

I find out that it is a complex problem to achieve same style as we have for the `Cardholder Name` input field. Hosted fields render is in an iframe and PayPal only allow few style properties to implement on fields. 
To resolve this issue we can apply a few properties to hosted field containers like `border`, `border-radius` etc. and rest to the hosted field itself.  `height` is difficult to figure out dynamically that why we have to set it manually in CSS.

Commits: https://github.com/impress-org/givewp/compare/8397a831e245...49e433105787
Slack discussion: https://givewp.slack.com/archives/C015XD31J5S/p1596649282001100

Updated screenshot:


Multi step form template:
image


Legacy form template:
image

**Comment by ravinderk** [comment]
@JasonTheAdams I find out that separator does not look good on `Multi step form` template. I will fix it.

**Comment by ravinderk** [comment]
Separator style updated to support legacy and multi step form template.

image
image

**Comment by ravinderk** [comment]
@JasonTheAdams I saw the need for `give_paypal_commerce_hosted_field_style` where users want to edit hosted field style if javascript logic fails to update it correctly.

you can hook into this filter to apply custom style:

```
<?php
add_filter( 'give_paypal_commerce_hosted_field_style', 'paypal_hosted_field_style' );
function paypal_hosted_field_style( $styles ) {
    $styles[':focus']['color'] = 'red';

    return $styles;
} 

```

you have to pass the value in the same format as PayPal expects.
image

**Comment by ravinderk** [comment]
@JasonTheAdams this pr is ready to review

File: assets/src/js/frontend/paypal-commerce/AdvancedCardFields.js [hunk 2]
```diff
@@ -253,6 +305,105 @@ class AdvancedCardFields extends PaymentMethod {
...
+		} );
+	}
+
+	/**
+	 * Set style properties for hosted card field and its container.
+	 *
+	 * @since 2.8.0
+	 */
+	setStyles() {
+		const self = this;
```
[the remark this answers is not in the document]
Comment (ravinderk) [reply assets/src/js/frontend/paypal-commerce/AdvancedCardFields.js]: fixed: https://github.com/impress-org/givewp/compare/66c49e14f055...29fc47c8a2f3

**Comment by ravinderk** [comment]
- PayPal smart button has `max-width`, so I can not make them full width. I experiment with making them center but that does not look good.

image



- The height of the input fields looks good on themes.

Twenty twenty theme
image

Twenty sixteen theme
image

File: src/PaymentGateways/PayPalCommerce/AdvancedCardFields.php [hunk 3]
```diff
@@ -184,16 +181,29 @@ private function cardExpirationField( $idPrefix ) {
...
+	 * @return string
+	 */
+	private function getSeparator(){
+		$label = esc_html__( 'or pay with card', 'give' );
+
+		ob_start();
+		include GIVE_PLUGIN_DIR . 'src/Views/Utils/separator-with-center-text.php';
+
+		return ob_get_clean();
+	}
```
[the remark this answers is not in the document]
Comment (ravinderk) [reply src/PaymentGateways/PayPalCommerce/AdvancedCardFields.php]: fixed: https://github.com/impress-org/givewp/compare/e030a5479d86...bf4b1a1805d8

**Comment by ravinderk** [comment]
@JasonTheAdams This pr is ready to review again

**Pull request merged**